### PR TITLE
✨ [amp-autocomplete] Add support for `prefetch` attribute for prefetching remote data

### LINKED ISSUE #24687: amp-autocomplete: Support prefetching remote data via an attribute to improve responsiveness for users
Issue to track followup effort for [#23772 (comment)](https://github.com/ampproject/amphtml/issues/23772#issuecomment-519284625) and [#24645 (comment)](https://github.com/ampproject/amphtml/pull/24645#pullrequestreview-291321195)

Add an amp-autocomplete attribute to support prefetching remote data since the default is to fetch on `touchstart` or input `focus`, whichever occurs first.

### PR DESCRIPTION
**Pull request by dmanek**
1. Implemented support for `prefetch` attribute
1. Minor refactoring
1. Added unit test

Resolves #24687

### CODE REVIEW

**Comment by dmanek** [comment]
cc @caroqliu

**Review by caroqliu — COMMENTED** [review]
LGTM, however note that users cannot start using this attribute until it is allowed in the [validation rules](https://github.com/ampproject/amphtml/blob/master/extensions/amp-autocomplete/validator-amp-autocomplete.protoascii). Please allow `prefetch` and update the validator tests too. :)

**Comment by dmanek** [comment]
Thanks for your comments @caroqliu. I've updated the validation rules and added tests, PTAL.

**Review by caroqliu — APPROVED** [review]
Thank you! Looks good. Please also add this attribute to the documentation in `extensions/amp-autocomplete/amp-autocomplete.md`

**Comment by dmanek** [comment]
Thanks for approving. I've already updated the documentation in `amp-autocomplete.md` 🙂

File: extensions/amp-autocomplete/validator-amp-autocomplete.protoascii [hunk 1]
```diff
@@ -87,6 +87,7 @@ tags: {  # <amp-autocomplete>
   attrs: {
     name: "[src]"
   }
+  attrs: { name: "prefetch" }
```
Comment (honeybadgerdontcare) [inline extensions/amp-autocomplete/validator-amp-autocomplete.protoascii:90]: please alphabetize by attribute name (e.g. this should go before `query`)

File: extensions/amp-autocomplete/validator-amp-autocomplete.protoascii [hunk 2]
```diff
@@ -131,6 +132,7 @@ tags: {  # <amp-autocomplete>
     name: "template"
     value_oneof_set: TEMPLATE_IDS
   }
+  attrs: { name: "prefetch" }
```
Comment (honeybadgerdontcare) [inline extensions/amp-autocomplete/validator-amp-autocomplete.protoascii:135]: same here

**Review by honeybadgerdontcare — CHANGES_REQUESTED** [review]

**Comment by caroqliu** [comment]
> Thanks for approving. I've already updated the documentation in `amp-autocomplete.md` 🙂

🤦‍♀️  You're right! Ignore me :)

**Review by honeybadgerdontcare — APPROVED** [review]

**Review by zhouyx — APPROVED** [review]

**Pull request merged**